ExecuteJobs.java

```diff
@@ -0,0 +1,124 @@
+import java.util.stream.*;
+import java.util.*;
+import java.io.*;
+//import java.util.Stream;
+
+public class ExecuteJobs {
+  private static final int ERR_SUCCESS = 0;
+  private static final int ERR_FAILED_TO_KILL = 1;
+  private static final int ERR_FAILED_TO_FIND = 2;
+  private static final int ERR_FAILED_TO_START = 3;
+
+  public ErrorInfo start(String sProcess) throws IOException {
+    ErrorInfo errorInfo = new ErrorInfo();
+    Process process = Runtime.getRuntime().exec(sProcess);
+
+    ErrorInfo errorStatus = getProcessStatus(process.pid());
+    if(errorStatus.getIoMessage() == null) {
+      errorInfo.setErrorCode(ERR_FAILED_TO_START);
+      errorInfo.setIoMessage("Failed to start " + sProcess);
+      return errorInfo;
+    }
+    errorInfo.setErrorCode(ERR_SUCCESS);
+    errorInfo.setIoMessage("Successfully start " + sProcess);
+    errorInfo.setPid(process.pid());
+    return errorInfo;
```

Comment:
Have you considered using exceptions instead of a custom error structure and returning just the process id for success?
I could imagine something like this would be more readable:
```java
public Pid start(String cmdline) throws ProcessException {
    try {
        Process p = startProcess(cmdline);
        // With no exceptions thrown, it is implicitly ERR_SUCCESS
        return p.pid();
    } catch (IOException exception) {
        throw ProcessException(exception.getMessage());
    }
}
```
and have `ProcessException` for cases where you need to relay additional error metadata.

Reply:
I did not. I wanted start, stop, etc to all return standardized error messages. Returning pid as success does not make sense for all of the methods in this class. Additionally, in some cases I want to have more verbose success/error messages than just the pid.